graphgrid_sdk/ggcore/nmt_train_pipeline.py

```diff
@@ -0,0 +1,150 @@
+                return new_model
+            else:
+                return active_model
+
+        else:
+            # With no training or eval metrics choose more recent models
+            return new_model
+    else:
+        # Properties do not match. Choose the newer model
+        return new_model
+
+
+class NmtTrainPipeline:
+    _configuration: SdkBootstrapConfig
+
+    _config_client: ConfigClient
+    _nlp_client: NlpClient
+
+    def __init__(self, bootstrap_config: SdkBootstrapConfig):
+        self._configuration = bootstrap_config
+
+        self._setup_clients()
+
+    def _setup_clients(self):
+        """Setup low-level clients."""
+        self._config_client = ConfigClient(self._configuration)
+        self._nlp_client = NlpClient(self._configuration)
+
+    def nmt_train_pipeline(self, models_to_train, datasets, no_cache, gpu, autopromote, success_handler,
+                           failed_handler):
+
+        num_models = len(models_to_train)
+
+        train_request_bodies = []
+        for model in models_to_train:
+            train_request_bodies.append(
+                TrainRequestBody(model=model, datasets=datasets, no_cache=no_cache, gpu=gpu))
+
+        nmt_train_responses = []
+        for i in range(num_models):
+            nmt_train_responses.append(self._nlp_client.trigger_nmt(train_request_bodies[i]))
+
+        nmt_status_responses = []
+        for i in range(num_models):
+            nmt_status_responses.append(self._nlp_client.get_nmt_status(nmt_train_responses[i].dagRunId))
+
+        completed_runs = 0
+        while completed_runs < num_models:
+            print("...running dag...")
+            time.sleep(10)
+            for status in nmt_status_responses:
+                if status.state != "success" and status.state != "failed":
+                    status_idx = nmt_status_responses.index(status)
+                    nmt_status_responses[status_idx] = self._nlp_client.get_nmt_status(
+                        nmt_train_responses[status_idx].dagRunId)
+                    if nmt_status_responses[status_idx].state == "success" or \
+                            nmt_status_responses[status_idx].state == "failed":
+                        completed_runs = completed_runs + 1
+
+        for model_status in nmt_status_responses:
+            if model_status.state == "success" and success_handler is not None:
+                success_handler(model_status)
+            if model_status.state == "failed" and failed_handler is not None:
+                failed_handler(model_status)
+
+        print("Dag training/eval/model upload has finished.")
+
+        promoted_models = []
+        if autopromote:
+            for i in range(num_models):
+                status = nmt_status_responses[i]
+                if status.state == "success":
+                    selected_model = evaluate_models(self._nlp_client.get_active_model(models_to_train[i]), status)
```

Comment:
get_active_model takes "task name" (e.g. named_entity_recognition), but right now we are giving it "model type" (e.g. "named-entity-recognition"). An error will be thrown here until there is alignment with model type/task